### PR TITLE
Fixed tag existence check and improved error reporting and handling

### PR DESCRIPTION
**Pull request by jamesbrown7**

### CODE REVIEW

**Comment by dasGitHub** [comment]
:+1:

File: examples/CSVImporter/CSVImporter.go [hunk 1]
```diff
@@ -199,11 +215,10 @@ func ProcessCSVRecord(jc jcapi.JCAPI, userList []jcapi.JCUser, csvRecord []strin
 
 			// Create or modify the tag in JumpCloud
 			tempTag.Id, err = jc.AddUpdateTag(opCode, tempTag)
-
 			if err != nil {
-				fmt.Printf("Could not process tag '%s', err='%s'", tempTag.ToString(), err)
+				err = fmt.Errorf("Could not POST tag '%s', err='%s'", tempTag.ToString(), err)
 			} else {
-				fmt.Printf("Processed tag ID '%s'\n", tempTag.Id)
+				fmt.Printf("\tCreated tag '%s' (ID '%s')\n", tempTag.Name, tempTag.Id)
```
Comment (kusold) [inline examples/CSVImporter/CSVImporter.go]: We have this in a few places, but why `fmt.Printf("\t.....")`? Wouldn't we want logs on a new line?

**Comment by kusold** [comment]
:+1: -- My comment on the Printf is more for my understanding than anything.

File: examples/CSVImporter/CSVImporter.go (hunk 1 above)
Comment (jamesbrown7) [inline examples/CSVImporter/CSVImporter.go]: They are on a new-line, and it looks like:

Line #1:
   User 'mlansky' (ID '553028cd7bb78c4a1103f559') updated from input file
   Tag 'Win81Dev - Meyer Lansky (mlansky)' already exists, not modifying it.
Line #2:
   User 'vito' (ID '553028cf00678e380b03f2f6') updated from input file
   Tag 'Win81Dev - Vito Corleone (vito)' already exists, not modifying it.

The tabs didn't carry through, but "User' and "tag" lines are indented...

File: examples/CSVImporter/CSVImporter.go [hunk 2]
```diff
@@ -137,25 +134,38 @@ func ProcessCSVRecord(jc jcapi.JCAPI, userList []jcapi.JCUser, csvRecord []strin
...
-		fmt.Printf("Processed user ID '%s'\n", currentUserId)
+		if opCode == jcapi.Update {
+			fmt.Printf("\tUser '%s' (ID '%s') updated from input file\n", currentUser.UserName, currentUserId)
+		} else {
+			fmt.Printf("\tLoaded user '%s' (ID '%s')\n", currentUser.UserName, currentUserId)
+		}
 	}
 
 	// Create/associate JumpCloud tags for the host and user...
 	if currentHost != "" {
```
Comment (topherMarie) [inline examples/CSVImporter/CSVImporter.go:156]: changing this to 
if currentHost == ""{
  return;
}
would eliminate 75 lines of indentation

**Pull request merged**